app/furniture/marketplace/checkout.rb

```diff
@@ -37,6 +38,11 @@ def create_stripe_session(success_url: , cancel_url: )
       })
     end
 
+    def complete(stripe_session_id:)
+      update!(status: :paid, stripe_session_id: stripe_session_id)
+      cart.update!(status: :checked_out)
+    end
```

Comment:
I figured shoveling this in here is probably better than leaving it in the controller; but maybe I'm wrong.

Reply:
I think it's okay to leave the `complete` method here for now. It's just updating the cart record right? It's not doing anything in request/response flows -- which is what the checkouts_controller is for right?

Reply:
Yep, it's doing the data transformations; so the controller just happens to know that showing a checkout when stripe gives it an id it also completes the checkout.

Reply:
I like to leave conversations open, rather than "resolving" them. Resolving feels like it dismisses the conversation! You had good questions! Other people may have them too!